cumulusci/robotframework/pageobjects/ObjectManagerPageObject.py

```diff
@@ -0,0 +1,239 @@
+            message="Detail page did not load in 1 min",
+        )
+
+    def create_lookup_field(self, field_name, related):
+        """Creates a Lookpup field by taking in the inputs field_name and related field"""
+        option = object_manager["select_related_option"].format(related)
+        related = object_manager["select_related"].format("DomainEnumOrId")
+        self.selenium.wait_until_page_contains_element(lookup_locator, 60)
+        self.selenium.click_element(lookup_locator)
+        self.selenium.wait_until_page_contains_element(next_button, 60)
+        self.selenium.click_element(next_button)
+        self.selenium.wait_until_page_contains_element(related, 60)
+        self.selenium.scroll_element_into_view(related)
+        self.selenium.get_webelement(related).click()
+        self.selenium.click_element(option)
+        self.selenium.wait_until_page_contains_element(next_button, 60)
+        self.selenium.click_element(next_button)
+        self.salesforce.populate_field("Field Label", field_name)
+        self.salesforce.populate_field(
+            "Description", "This is a custom field generated during automation"
+        )
+        self.selenium.click_element(next_button)
+        self.selenium.click_element(next_button)
+        self.selenium.click_element(next_button)
+        self.selenium.click_element(save_button)
+        self.selenium.wait_until_location_contains(
+            "FieldsAndRelationships/view",
+            timeout=90,
+            message="Detail page did not load in 1 min",
+        )
+
+    @capture_screenshot_on_error
+    def is_field_present(self, field_name):
+        """Verifies and asserts the field got created """
+        self.selenium.wait_until_page_contains_element(search_button, 60)
+        self.selenium.get_webelement(search_button).send_keys(field_name)
+        self.selenium.get_webelement(search_button).send_keys(Keys.ENTER)
+        time.sleep(1)
+        self.salesforce.wait_until_loading_is_complete()
+        search_results = object_manager["search_result"].format(field_name)
+        assert (
+            len(self.selenium.get_webelements(search_results)) == 1
+        ), "Expected result count to be 1 but found {}".format(
+            len(self.selenium.get_webelements(search_results))
+        )
+
+    @capture_screenshot_on_error
+    def delete_custom_field(self):
+        """Deletes the added custom field"""
+        self.selenium.click_element(actions_menu)
+        self.selenium.wait_until_page_contains_element(action_item_delete, 60)
+        self.selenium.click_element(action_item_delete)
+        self.selenium.wait_until_page_contains_element(confirm_delete, 60)
+        self.selenium.click_element(confirm_delete)
+        self.selenium.wait_until_location_contains("/view", timeout=90)
+
+    @capture_screenshot_on_error
+    def create_custom_field(self, **kwargs):
+        """Ensure that the custom field does not exist prior and Creates a custom field based
```

Comment:
This description doesn't sound quite right. If the field already exists, what does this keyword do? Will it throw an error, or will it silently return since the field already exists? The documentation implies it throws an error ("Ensures that...") but it looks like it just returns if it already exists.

Would it be better to say "this creates a custom field if one doesn't already exist by the given name"?

This documentation should also include an example, since it's not clear just from the documentation what "kwargs" should be. An example would quickly clear up any misunderstanding.

Maybe something like:

```
Example:

| Create custom field
| ...  Object=Payment
| ...  Field_Type=Formula
| ...  Field_Name=Is Opportunity From Prior Year
| ...  Formula=YEAR( npe01__Opportunity__r.CloseDate ) < YEAR( npe01__Payment_Date__c )

Reply:
updated